src/common/perf_counters_collection.cc

```diff
@@ -0,0 +1,64 @@
+#include "common/perf_counters_collection.h"
+#include "common/ceph_mutex.h"
+#include "common/ceph_context.h"
+
+/* PerfcounterCollection hold the lock for PerfCounterCollectionImp */
+PerfCountersCollection::PerfCountersCollection(CephContext *cct)
+  : m_cct(cct),
+    m_lock(ceph::make_mutex("PerfCountersCollection"))
+{
+  perf_impl = new PerfCountersCollectionImpl();
```

Comment:
i'd suggest use `unique_ptr<>`.

Reply:
also https://github.com/ceph/ceph/pull/24141#discussion_r220123071 . again, why can't we simply have `PerfCountersCollectionImpl` as a member variable?